undertow/src/main/java/org/wildfly/extension/undertow/deployment/UndertowJSRWebSocketDeploymentProcessor.java

```diff
@@ -196,7 +197,11 @@ private void bindJndiServices(final DeploymentUnit deploymentUnit, final Service
         webSocketInfo.addListener(new WebSocketDeploymentInfo.ContainerReadyListener() {
             @Override
             public void ready(ServerWebSocketContainer container) {
-                binderService.getManagedObjectInjector().inject(new ImmediateManagedReferenceFactory(container));
+                try {
+                    binderService.getManagedObjectInjector().inject(new ImmediateManagedReferenceFactory(container));
```

Comment:
This needs to use a listener to uninject on undeploy, otherwise you end up with a stale container object from the previous deployment.